scripts/tslint/noNonPrimitiveNamesRule.ts

```diff
@@ -0,0 +1,42 @@
+import * as Lint from "tslint/lib/lint";
+import * as ts from "typescript";
+
+export class Rule extends Lint.Rules.AbstractRule {
+    public static get_replacement(name: string): string {
+        var replacement: string;
+        switch (name) {
+            case 'Number':
+                replacement = 'number';
+                break;
+            case 'Boolean':
+                replacement = 'boolean';
+                break;
+            case 'Object':
+                replacement = '{}';
+                break;
+            case 'String':
+                replacement = 'string';
+                break;
+        }
+        return replacement;           
+    }
+    public static FAILURE_STRING_FACTORY = (name: string, replacement: string) => `Use of non primitive name '${name}' is not allowed - use '${replacement}' instead`;

```

Comment:
It's a pattern across all the other tslitnt files as well like [this](https://github.com/Microsoft/TypeScript/blob/master/scripts/tslint/noIncrementDecrementRule.ts). Do you think we should update it?